packages/clay-core/src/tree-view/TreeView.tsx

```diff
@@ -0,0 +1,71 @@
+/**
+ * SPDX-FileCopyrightText: © 2021 Liferay, Inc. <https://liferay.com>
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+
+import classNames from 'classnames';
+import React from 'react';
+
+import {TreeViewGroup} from './TreeViewGroup';
+import {TreeViewItem, TreeViewItemStack} from './TreeViewItem';
+import {TreeViewContext} from './context';
+import type {IExpandable, IMultipleSelection} from './useTree';
+
+type ChildrenFunction<T> = (item: T) => React.ReactElement;
+
+interface ITreeViewProps<T>
+	extends React.HTMLAttributes<HTMLUListElement>,
+		IMultipleSelection,
+		IExpandable {
+	children: React.ReactNode | ChildrenFunction<T>;
+	displayType?: 'light' | 'dark';
+	items?: Array<T>;
+	showExpanderOnHover?: boolean;
+}
+
+export function TreeView<T>(
+	props: ITreeViewProps<T>
+): JSX.Element & {
+	Item: typeof TreeViewItem;
+	Group: typeof TreeViewGroup;
+	ItemStack: typeof TreeViewItemStack;
+};
+
+export function TreeView<T>({
+	children,
+	className,
+	displayType = 'light',
+	items,
+	showExpanderOnHover = true,
+	...otherProps
+}: ITreeViewProps<T>) {
```

Comment:
Unlike type on the other components in this one we are stopping using `React.FunctionComponent` as I had commenting in #4054. For more information on why, read also https://github.com/facebook/create-react-app/pull/8177